src/spatialdb.cpp

```diff
@@ -44,7 +44,7 @@ const vector<string> DatabaseInitCommands = {
 
     "CREATE TABLE IF NOT EXISTS services ( "
     "  nodeId       TEXT NOT NULL, "
-    "  serviceType  INT NOT NULL, "
+    "  serviceType  TEXT NOT NULL, "
```

Comment:
I see you have properly changed SQL variable binding types already, so the DB should work now.
But without this change I doubt that it was properly working before, did you perform any tests already?